src/x/cache/lru_cache_test.go

```diff
@@ -584,6 +584,21 @@ func TestLRU_PutWithTTL_NoExistingEntry(t *testing.T) {
 	assert.Equal(t, "bar", value.(string))
 }
 
+func TestLRU_GetNonExisting_FromFullCache_AfterDoublePut(t *testing.T) {
+	lru := NewLRU(&LRUOptions{MaxEntries: 2})
+
+	// insert same key twice:
+	lru.Put("foo", "1")
+	lru.Put("foo", "1")
+
+	// insert another key to make cache full:
+	lru.Put("bar", "2")
+
+	// try to get entry that does not exist:
```

Comment:
it might nice to document the regression (infinite loop) that this test is protecting against